src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -121,13 +121,16 @@ public static IEnumerable<ITypeSymbol> GetBaseTypesAndThis(this ITypeSymbol? typ
         }
     }
 
-    public static IEnumerable<INamedTypeSymbol> GetBaseTypes(this ITypeSymbol type)
+    public static IEnumerable<INamedTypeSymbol> GetBaseTypes(this ITypeSymbol? type)
     {
-        var current = type.BaseType;
-        while (current != null)
+        if (type is not null)
         {
-            yield return current;
-            current = current.BaseType;
+            var current = type.BaseType;
```

Comment:
would `type?.BaseType` work here instead of the if?

Reply:
yes. done.